README.md

```diff
@@ -1,2 +1,133 @@
-Dapr SDK to compose pluggable components in Java.
+
+[Dapr] SDK to compose Pluggable components in Java. Also defines some sample components.
+
+For further information on Pluggable Components, check:
+*  [(Proposal) (Updated) gRPC Components (aka "Pluggable components") · Issue #4925](https://github.com/dapr/dapr/issues/4925)
+*  [Pluggable Components QuickStart]
+
+# Getting started
+
+## Pre-Requisites
+
+We build on [Dapr Java SDK's dependencies](https://github.com/dapr/java-sdk/blob/master/README.md#getting-started) and
+additionally require the following:
+
+* [Gradle 7.x](https://gradle.org/install/)
+* [Docker](https://docs.docker.com/engine/install/)
+
+## Developing new components
+
+The code under `src/main/java/org/example/` exemplifies how one can
+
+1. Define new pluggable components.
+
+   In that directory, check [InMemoryStateStore.java] for an example
+   of a simple StateStore component that stores all its data in memory.
+
+   As a pluggable component developer, most if not all of your work will be
+   spent on this step. We defined Java-like interfaces so you don't have to worry
+   about the tedious work of translating gRPC classes and model to and
+   from Java.
+
+2. Turn them into servers capable extending Dapr functionality.
+
+   Once a component is defined, we wrap them with the necessary code to
+   turn them into servers ready to take requests from Dapr. You can take a 
+   look on [StateStoreComponentServer.java] for an example of how we did that
+   for [InMemoryStateStore.java].
+
+   We also took care of all the tedious work for step 2 by defining a wrapper
+   that handles all the tedious work of handling command line parsing,
+   setting up a server listening on a Unix Socket Domain etc.
+
+TODO: talk about service discovery and how we tie a running component server to
+a component configuration as `config/container_java_memstore.yaml`.
+
+# Compiling and running the code
+
+The easiest way to get up and running is using docker-compose. 
+
+## Running using docker-compose
+
+Running using `docker-compose` allows to quickly test the whole solution with no compilation required. It also portrays how one must orchestrate the start of pluggable components and `daprd` itself.
+
+In the current directory run:
+
+```bash
+docker-compose up
+```
+
+If you rather have this ran in detached mode, add  `-d` parameter at the end of that command.
+
+You can run the commands from the Testing section bellow after starting running `docker-compose`.
+
+
+## Running as a Docker image
+
+Another option is to build a Docker image for the component and run it from there.
+
+You might want to take this route once your component is stable and is ready to
+test with your local Dapr instance.
+
+Build and start the container:
+
+```
+docker build -t javacomponent .
+mkdir -pv  /tmp/sharedUDS
+docker run -e DAPR_COMPONENT_SOCKET_PATH=/tmp/sharedUDS/javaMemstore.socket -v /tmp/sharedUDS/:/tmp/sharedUDS/ javacomponent
```

Comment:
Are we doing this for other SDKs? socket.sock seams a rather generic name. Can't we use something a bit more descriptive or self-documenting?

Reply:
Thinking about it, this will have to be updated anyway, as the UDS path is not arbitrary, right? It is deterministic based on the .YAML configuration for the component.